src/ptvsd/_vendored/pydevd/_pydevd_bundle/pydevd_process_net_command_json.py

```diff
@@ -663,5 +664,48 @@ def on_goto_request(self, py_db, request):
         response = pydevd_base_schema.build_response(request, kwargs={'body': {}})
         return NetCommand(CMD_RETURN, 0, response, is_json=True)
 
+    def _can_set_dont_trace_pattern(self, py_db, start_patterns, end_patterns):
+        if py_db.is_cache_file_type_empty():
```

Comment:
Actually, as long as it's empty I think it should be ok to set right? (so, it should return as `success` if empty or if it's the same thing which was already before).

Reply:
For clarification:
* If cache is not empty we should fail the don't trace request.
* If cache is empty we show allow setting the patterns.

Is there any need to check the patterns at all?

Reply:
Humm, not really... I was thinking about something as:

```
if cache is empty:
    return success=True

if pattern was set and new pattern is still the same: 
    return success=True  (I think this can happen when reconnecting)

return success=False
```